src/Lecture4.hs

```diff
@@ -180,14 +210,30 @@ The 'Stats' data type has multiple fields. All these fields have
 instance for the 'Stats' type itself.
 -}
 
+strictMayBeAppend :: (Semigroup a) =>  Maybe a -> Maybe a -> Maybe a
+strictMayBeAppend (Just !x) (Just !y) = Just $ x<>y
+strictMayBeAppend (Just !x) Nothing  = Just x
+strictMayBeAppend Nothing (Just !y)  = Just y
+strictMayBeAppend Nothing Nothing  = Nothing 
```

Comment:
That's absolutely correct implementation 👍🏻 
You can write it a bit shorter though by pattern matching only on the final result:

```suggestion
strictMayBeAppend :: (Semigroup a) =>  Maybe a -> Maybe a -> Maybe a
strictMayBeAppend a b = case a <> b of
    Nothing -> Nothing
    Just !x -> Just x
```